src/main/java/com/google/codeu/servlets/StatsServlet.java

```diff
@@ -0,0 +1,52 @@
+package com.google.codeu.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.codeu.data.Datastore;
+
+import java.io.IOException;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * A servlet for handling fetching site statistics.
+ */
+@WebServlet("/stats")
+public class StatsServlet extends HttpServlet{
+
+  private Datastore datastore;
+
+  @Override
+  /**
+   * Called automatically once when the servlet is first created,
+   * to create and store a Datastore instance.
+   */
+  public void init() {
+    datastore = new Datastore();
+  }
+
+  @Override
+  /**
+   * On page /stats, gets total message count from the datastore and outputs it
+   * as bare-bones JSON format: {"messageCount": messageCount}.
+   */
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+    throws IOException, ServletException {
+
+    UserService userService = UserServiceFactory.getUserService();
+
+    boolean isUserLoggedIn = userService.isUserLoggedIn();
+    request.setAttribute("isUserLoggedIn", isUserLoggedIn);
```

Comment:
I noticed that you're doing this in a few different servlets. You could consider moving this into a shared location. For something simple, you could use a static function in a utility class? For a more advanced solution, you could look into servlet filters.

That doesn't have to be a part of this PR though.

Reply:
Mm I'll look into that.